rest/src/main/scala/io/udash/rest/raw/RestResponse.scala

```diff
@@ -95,3 +112,154 @@ trait RestResponseLowPrio { this: RestResponse.type =>
     implicit forBody: ImplicitNotFound[AsRaw[HttpBody, T]]
   ): ImplicitNotFound[AsRaw[RestResponse, T]] = ImplicitNotFound()
 }
+
+/**
+ * Streaming REST response containing a status code, headers, and a streamed body.
+ * Unlike standard [[RestResponse]], the body content can be delivered incrementally through a reactive stream.
+ */
+final case class StreamedRestResponse(
+  code: Int,
+  headers: IMapping[PlainValue],
+  body: StreamedBody,
+) extends AbstractRestResponse {
+
+  def header(name: String, value: String): StreamedRestResponse =
+    copy(headers = headers.append(name, PlainValue(value)))
+
+  def ensureNonError: StreamedRestResponse =
+    if (isSuccess) this else throw HttpErrorException(code, StreamedBody.toHttpBody(body))
+}
+
+object StreamedRestResponse extends StreamedRestResponseLowPrio {
+
+  /**
+   * Converts a [[StreamedRestResponse]] to a standard [[RestResponse]] by materializing streamed content.
+   * This is useful for compatibility with APIs that don't support streaming.
+   */
+  def fallbackToRestResponse(response: StreamedRestResponse): Task[RestResponse] = {
+    val httpBody: Task[HttpBody] = response.body match {
+      case StreamedBody.Empty =>
+        Task.now(HttpBody.Empty)
+      case binary: StreamedBody.RawBinary =>
+        Utils.mergeArrays(binary.content).map(HttpBody.Binary(_, binary.contentType))
+      case jsonList: StreamedBody.JsonList =>
+        jsonList.elements
+          .foldLeftL(new StringBuilder("[")) { case (sb, json) =>
```

Comment:
`StringBuilder`s are not safe for use with multiple threads. I don't think we're guaranteed it's the same thread here, only that it's one at a time.